uint16 distance
---
string message
